### PR TITLE
Update omniauth workflow to store age rather than birthday.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Is there a good way to test this? Though I assume `auth.info.dob` is in the form `%Y-%m-%d` (not seeing any transformation and given that is the format in our DB), I don't know the best way to confirm this for our myriad of oauth providers.

EDIT: Looking at the test (see https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/controllers/omniauth_callbacks_controller_test.rb#L206), it appears to be a ruby `Date` object. Making appropriate adjustments.

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -381,7 +381,8 @@ def self.name_from_omniauth(raw_name)
         user.age = 21
       else
         # student or unspecified type
-        user.birthday = auth.info.dob
+        age_in_years = ((DateTime.now - Date.parse(auth.info.dob, '%Y-%m-%d')) / 365).floor
```
Comment (ashercodeorg) [inline dashboard/app/models/user.rb]: Note that this slightly overestimates the age of a student, as the result of leap years. Using 365.25 could also be an over- or under- estimate. Consequently, I'm accepting slight error here.

If we want to make sure we never overestimate (at the risk of locking out a thirteen year old), let me know, I can change the math to do so.

**Comment by ashercodeorg** [comment]
CircleCI failure is unrelated. Ready for review.

File: dashboard/app/models/user.rb [hunk 2]
```diff
@@ -381,7 +381,7 @@ def self.name_from_omniauth(raw_name)
         user.age = 21
       else
         # student or unspecified type
-        user.birthday = auth.info.dob
+        user.age = ((Date.today - auth.info.dob) / 365).floor if auth.info.dob
```
Comment (jeremydstone) [inline dashboard/app/models/user.rb:384]: I verified this is the same calculation performed in User#age

**Comment by jeremydstone** [comment]
LGTM

**Pull request merged**